src/components/Composer/index.js

```diff
@@ -123,21 +113,38 @@ class Composer extends React.Component {
             ? `${props.defaultValue}`
             : `${props.value || ''}`;
 
+        this.selection = {
+            start: initialValue.length,
+            end: initialValue.length,
+        };
+
         this.state = {
             numberOfLines: 1,
-            selection: {
-                start: initialValue.length,
-                end: initialValue.length,
-            },
+            value: initialValue,
         };
         this.dragNDropListener = this.dragNDropListener.bind(this);
         this.paste = this.paste.bind(this);
         this.handlePaste = this.handlePaste.bind(this);
         this.handlePastedHTML = this.handlePastedHTML.bind(this);
         this.handleWheel = this.handleWheel.bind(this);
+        this.updateNumberOfLines = this.updateNumberOfLines.bind(this);
+        this.onChangeText = this.onChangeText.bind(this);
+        this.focus = this.focus.bind(this);
+        this.onSelectionChange = this.onSelectionChange.bind(this);
+        this.updateSelection = this.updateSelection.bind(this);
     }
 
     componentDidMount() {
+        // we pass the ref to the native view instance,
+        // however, we want this method to be
+        // available to be called from the outside as well.
+        this.textInput.onChangeText = this.onChangeText;
+        this.textInput.updateSelection = this.updateSelection;
+
+        // overwrite focus with this components implementation
+        this.textInput.nativeFocus = this.textInput.focus;
+        this.textInput.focus = this.focus;
+
```

Comment:
So, we have a `this.textinput` check at line 160. That seems unnecessary if this is working fine.